### ISSUE TITLE
User unable to be deleted after 0.90 -> 0.90.1 upgrade

### ISSUE THREAD

**Issue by aldahick** [glpi-project/glpi#298]
I am experiencing an issue on GLPI 0.90.1 (immediately following the update) in which one of our technicians is unable to be removed from the ticket. As well, his name will normally show up - on this ticket alone, only his email appears. My name and status are fine, as is the requester. However, I cannot delete anyone from the ticket. If there are logs I should provide, please let me know.

![image](https://cloud.githubusercontent.com/assets/2233316/11539417/e72b974a-98f4-11e5-875c-47f1bce5aa15.png)

**Comment by orthagh** [comment]
Yes, logs could help

you could find them in glpi/files/_logs/

**Comment by yllen** [comment]
What is the status of the ticket?

**Comment by Tiin57** [comment]
After review, the logs show nothing except routine logins and events.
The ticket status at the time of the issue was "Processing (Assigned)".

**Comment by Tiin57** [comment]
It seems that the problem is created when we set Assigned To to a technician (only after assigning a Requester), and then put it back to ---. When it is reset back to no one assigned, it auto-fills the email of the user who is signed in. That field can be changed, but if the ticket is created without changing that field, it creates the problem above. By doing this, you can replicate the issue.

**Comment by loule** [comment]
I can confirm we have the same "issue". And like many of you, few techs think about changing email field after assigned field... And when you want to delete it after that, clicking the cross button, there's a permissions problem.

**Comment by yllen** [comment]
I can't reproduce.
I have a user without email. I assign him to the ticket. I open the assign part to change his email.
I can delete the user assign to the ticket

**Comment by loule** [comment]
The problem comes when email is already configured, so it's automatically changed when ticket is assigned.

**Comment by yllen** [comment]
I done another test with user with predefined email without problem.

**Comment by loule** [comment]
Hope it's not a random problem, or will be hard to find...
Did you do the 0.90.1 upgrade from 0.90?

**Comment by yllen** [comment]
i'm in 0.91svn since 0.78 version with all upgrades

**Comment by Tiin57** [comment]
The issue only arises when a requester is assigned during the creation of a ticket, then the above steps are followed. It does not appear to occur during the edit of an existing ticket, or during the creation of a ticket with no requester.

**Comment by loule** [comment]
Didn't see all cases, and look at title, its 0.90 -> 0.90.1  not 0.91 !  Maybe just mistyped, but could change a lot of things...

**Comment by Tiin57** [comment]
Yes, the upgrade was from 0.90 to 0.90.1, not to 0.91.

**Comment by yllen** [comment]
I make another test in 0.90.1 and can reproduce.
Can you indicate step by step all yours actions?

**Comment by loule** [comment]
can or can't reproduce? not sure what to understand :)
From the informations i have (from me and techs), when you create a new ticket, you select category or whatever, it then assign someone to the ticket automatically, with email also assigned.
When you select back empty "assigned to" (meaning that it is not yet assigned), the email doesn't get removed. If you forget to change back email too, you won't be able to remove it later.

Pretty hard to describe clearly....  Maybe Tiin57 could add some infos.

**Comment by Tiin57** [comment]
That seems accurate to me. If you need further info, I'd be happy to help.

**Comment by loule** [comment]
Hy all,
is the issue confirmed or is it something that we didn't understand?

**Comment by orthagh** [comment]
Hi all.

Sorry for the delay.
I just did the following tests (as you described) on a 0.90.1 and 0.90/bugfixes branch

* On existing, ticket, i select an "assign to" user
* Email field is auto filled
* I change the user to "--"
* In my case, email field is emptied.

And after save, no invalid user present in my actor bloc.
I also tested with new ticket form and still can't replicate issue.

If this issue is still present in your side, could you check the dev tools of your browser (F12 key), console tab, if any javascript error triggers ?

**Comment by loule** [comment]
The problem seems to appear when user is auto assigned

**Comment by yllen** [comment]
I can't reproduce even user is auto assigned in My settings

**Comment by orthagh** [comment]
@loule, @Tiin57 

Are you sure this strange behavior are not triggered by a plugin ?
Can you test disable all yours and retest ?

**Comment by loule** [comment]
already done it, was the same....
maybe a random setup bug.... we are just some of us impacted, will see with next version if better, it's not blocking

**Comment by orthagh** [comment]
see #756 . This issue did not trigger anymore in the 9.1 version coming soon
